B2g styling

shane-tomlinson opened the pull request:
@jedp - it needs cleanup yet, but here is basic B2G styling, with all the tests passing. Woot!

Commit:
Initial styling updates.

Commit:
Starting to get B2G styles in order.

Commit:
Set the dialog size to the B2G screen size.

Commit:
Start on the loading screen.

Commit:
Show a 'signing in' screen before generating the assertion

Commit:
Do not resize the screen on mobile devices smaller than the initial screen size. This prevents virtual keyboards from getting in the way.

Commit:
more styles. yay

Commit:
Loads up updates to buttons, labels and headers for mobile styling.

Commit:
Clean up the pick email screen to handle both mobile and desktop.

Commit:
Bring back the RP info.

Commit:
Add a loading screen. Add the loading spinner too.

Commit:
Show the "signing in" screen as soon as a user selects an email address.

Commit:
Woot! We have decent mobile styles, a footer that goes to the bottom of the screen, things are looking up.

Commit:
Style the error screen.

Commit:
fix up botched desktop version styles.

Commit:
Update the authentication page for the differences between mobile and desktop.

Commit:
Rotate the spinner using keyframe animations.

Commit:
Set the normal dialog size.

Commit:
Make the loading screen take up the entire screen on desktop. Center all text within it. Update spinner styles.

Commit:
Update the CSS selectors so the tests pass again.

Commit:
Get rid of the console message.

shane-tomlinson commented:
wait. jshint tests failing. I lied. pushing an update.

shane-tomlinson commented:
GitHub, you lie. 07194d0 has a green check from travis.

jedp commented on resources/static/common/js/modules/page_module.js:
```diff
@@ -24,6 +25,11 @@ BrowserID.Modules.PageModule = (function() {
 
+  function cancelDialog() {
+    /*jshint validthis: true*/
+    window.close();
```
This bit doesn't work on b2g.  Scripts may not close windows that were not opened by the script.  :(

jedp replied:
Spoke with Shane on IRC.  He suggested calling oncomplete with a null assertion, which used to be the cancel button in the old days.  This sounds good to me!

jedp commented on resources/static/dialog/css/style.css:
```diff
@@ -229,31 +237,98 @@ section > .contents {
+          animation-timing-function: linear;
+  margin-bottom: 10px;
+}
+
```
These changes are epic.  Amazing work, man.

shane-tomlinson commented on resources/static/common/js/wait-messages.js:
```diff
@@ -10,6 +10,10 @@ BrowserID.Wait = (function(){
       message:  gettext("In just a moment you'll be signed into Persona.")
     },
 
+    checkingEmail: {
```
This doesn't really belong here.

Commit:
Remove commented out CSS. Add comments for what the CSS is about.

Commit:
Fix Chrome for Android showing a blank screen.

* Remove the keyframe animation on the loading spinner.

Commit:
for cancelDialog, trigger the "cancel" message instead of window.close.

This will cause the success callback to be called with a null assertion. FirefoxOS friendly.

Commit:
Move the text for "checkingEmail" from wait-messages.js to dialog/js/modules.js

shane-tomlinson commented on resources/static/common/js/modules/page_module.js:
```diff
@@ -45,10 +53,11 @@ BrowserID.Modules.PageModule = (function() {
 
     stop: function() {
-      dom.removeClass("body", "waiting");
+      dom.removeClass("body", "loading");
```
Try putting this in start instead of stop.

Commit:
Only hide the wait screen when a new screen is shown, not when stopping a module.

Commit:
Use the black theme for "Is this your computer"

Commit:
Remove the commented out line.

shane-tomlinson commented on resources/static/dialog/js/misc/screen_size_hacks.js:
```diff
@@ -105,17 +113,35 @@
+        // Don't let the screen shrink to less than its initial height. This
+        // minimizes jerkiness associated with the virtual keyboard being
+        // displayed.
+        /*
```
Remove the commented out code here.

shane-tomlinson commented on resources/static/dialog/js/misc/screen_size_hacks.js:
```diff
@@ -69,8 +74,8 @@
-        if(maxEmailWidth > parentNaturalWidth) {
-          scrollableEl.css("width", parentNaturalWidth + "px");
+        if (maxEmailWidth > parentNaturalWidth) {
+          /*scrollableEl.css("width", parentNaturalWidth + "px");*/
```
Remove the commented out code here too.

Commit:
Clean up the screen_size_hacks

Commit:
Fix the forgotPassword selector so that Selenium runs.

Commit:
Fix the unwanted submit_disabled button spinner on mobile. Show the "confirm your email" screen in black.

Commit:
Make set password a bit easier.

If the user enters a password and then hits enter, as long as the vpassword field is empty, focus vpassword.

Commit:
Fix the buttonrow showing up in the "is this your computer" and "verify your email" screens when closing the dialog.

shane-tomlinson commented on resources/static/include_js/_include.js:
```diff
@@ -131,7 +131,7 @@
     var windowOpenOpts =
       (isFennec ? undefined :
-       "menubar=0,location=1,resizable=1,scrollbars=1,status=0,width=700,height=375");
+       "menubar=0,location=1,resizable=1,scrollbars=1,status=0,width=750,height=375");
```
wah wah.

Commit:
The dialog is only 700 px wide, not 750
